cairo_programs/cairo-1-programs/dict_non_squashed/dict_with_struct_non_squash.cairo

```diff
@@ -0,0 +1,28 @@
+use core::nullable::{nullable_from_box, match_nullable, FromNullableResult};
+
+
+#[derive(Drop, Copy)]
+struct FP16x16 {
+    mag: u32,
+    sign: bool
+}
+
+fn main() -> Felt252Dict<Nullable<FP16x16>> {
+    // Create the dictionary
+    let mut d: Felt252Dict<Nullable<FP16x16>> = Default::default();
+
+    let box_a = BoxTrait::new(identity(FP16x16 { mag: 1, sign: false }));
+    let box_b = BoxTrait::new(identity(FP16x16 { mag: 1, sign: true }));
+    let box_c = BoxTrait::new(identity(FP16x16 { mag: 1, sign: true }));
+
+    // Insert it as a `Span`
+    d.insert(0, nullable_from_box(box_c));
+    d.insert(1, nullable_from_box(box_a));
+    d.insert(2, nullable_from_box(box_b));
+
+    d
+}
+
+// TODO: remove this temporary fix once fixed in cairo
+#[inline(never)]
+fn identity<T>(t: T) -> T { t }
```

Comment:
An EOF newline is missing in some files. I suggest configuring your editor to add them automatically, as most unix utilities expect them: https://github.com/lambdaclass/lambdaclass_hacking_learning_path/blob/76538b6ae30b3563c006028abbe7dc2d5061005a/README.md?plain=1#L401-L403